rest/config.go

```diff
@@ -38,6 +38,7 @@ type (
 		MaxBytes int64  `json:",default=1048576"`
 		// milliseconds
 		Timeout      int64         `json:",default=3000"`
+		MaxTimeout   int64         `json:",default=-1"`
```

Comment:
Do we need an extra config item for this?

Reply:
I think if we need to use the `httpServer.ReadTimeout` and `httpServer.WriteTimeout` to ensure that the connection can be closed, then we should use the extra config `MaxTimeout` instead of the original `Timeout`.

Because the `Timeout` should be the default value, it should be overridden by the timeout within route. So if we use `Timeout` directly as a condition for forcing the connection to be closed, the timeout within route exceeding the global `Timeout` will be invalidated. -- https://github.com/zeromicro/go-zero/issues/2140#issuecomment-1185375190

If we need to relate httpServer timeout with global timeout and route timeout, then using a httpServer timeout may not be appropriate. Maybe we can hijack the connection in `rest/handler.(timeoutWriter)` to close it (I'm still thinking about how to implement it https://github.com/gitsang/go-zero/pull/1/files)